File: client/android/Contract/Navigation/Shell/MobileNavigationShellPayload.kt
package app.mobiling.client.contract.navigation.shell

data class MobileNavigationShellPayload(
    val schema: String,
    val channel: String,
    val platforms: List<String>,
    val locations: Map<String, List<MobileNavigationItemPayload>>,
) {
    fun items(location: String): List<MobileNavigationItemPayload> = locations[location].orEmpty()
}

data class MobileNavigationItemPayload(
    val key: String,
    val label: String,
    val icon: String?,
    val badge: String?,
    val enabled: Boolean,
    val visible: Boolean,
    val status: String,
    val disabledReason: String?,
    val requiredComponent: String?,
    val location: String,
    val group: String,
    val groupLabel: String,
    val action: String?,

File: client/ios/Contract/Navigation/Shell/MobileNavigationShellPayload.swift
import Foundation

public struct MobileNavigationShellPayload: Sendable {
    public let schema: String
    public let channel: String
    public let platforms: [String]
    public let locations: [String: [MobileNavigationItemPayload]]

    public func items(location: String) -> [MobileNavigationItemPayload] {
        locations[location] ?? []
    }
}

public struct MobileNavigationItemPayload: Identifiable, Sendable {
    public let id: String
    public let key: String
    public let label: String
    public let icon: String?
    public let badge: String?
    public let enabled: Bool
    public let visible: Bool
    public let status: String
    public let disabledReason: String?
    public let requiredComponent: String?
    public let location: String
    public let group: String
    public let groupLabel: String
    public let action: String?
    public let route: String?
